### ISSUE TITLE
Missing role name check in meta dependencies

### ISSUE THREAD

**Issue by netsandbox** [ansible/ansible-lint#3850]
##### Summary

Role names in meta files dependencies are not checked.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.20.3 using ansible-core:2.15.5 ansible-compat:4.1.10 ruamel-yaml:0.17.35 ruamel-yaml-clib:0.2.8
```

##### STEPS TO REPRODUCE

meta.yml
```console (paste below)
dependencies:
  - role: foo/bar
```

##### Desired Behavior

A meta.yml file with the above mentioned `dependencies` should produce a `Avoid using paths when importing roles.` warning.
##### Actual Behavior

No warning is shown.

**Comment by ajinkyau** [comment]
@netsandbox A PR to fix this would be more than welcomed. 
Thanks for reporting it.
